docs: Small fix and update of README.md (#740)
awslabs/syne-tune: README.md

RUSH | Zappella, et al. (2021)| random | yes | yes | yes 
BoundingBox | Perrone, et al. (2019) | any | yes | yes | yes

<sup>*</sup>: We implement the model-based scheduling logic of DyHPO, but use

> the same Gaussian process surrogate models than MOBSTER or HyperTune. The original

Reviewer: Same change: 
```suggestion
the same Gaussian process surrogate models as MOBSTER and HyperTune. The original
```
